README.rst

```diff
@@ -26,3 +26,371 @@ Run `tox` to run the whole test suite:
+connected the serial or ``DEBUG_ADAPTER_ID`` must be used to flash the
+correct device, and for some ``BOARD``\ s also the serial port ``PORT``.
+These variables can be appended to the environment of the spawned
+object, e.g:
+
+-  `FIT IoT-LAB <https://www.iot-lab.info/>`__:
+
+.. code:: python
+
+   # first device using dwm1001-1 on the saclay site
+   env1 = {'BOARD': 'dwm10001', 'IOTLAB_NODE': 'dwm1001-1.saclay.iot-lab.info'}
+   ctrl1 = RIOTCtrl(env=env1, application_directory='.')
+   # second device using dwm1001-2 on the saclay site
+   env2 = {'BOARD': 'dwm10001', 'IOTLAB_NODE': 'dwm1001-2.saclay.iot-lab.info'}
+   ctrl2 = RIOTCtrl(env=env2, application_directory='.')
+
+-  locally:
+
+.. code:: python
+
+   # first samr21-xpro
+   env1 = {'BOARD': 'samr21-xpro', 'DEBUG_ADAPTER_ID': 'ATML2127031800004957'}
+   ctrl1 = RIOTCtrl(env=env1, application_directory='.')
+   # second samr21-xpro
+   env2 = {'BOARD': 'samr21-xpro', 'DEBUG_ADAPTER_ID': 'ATML2127031800011458'}
+   ctrl2 = RIOTCtrl(env=env2, application_directory='.')
+
+For the advanced user one could also do as suggested in
+`multiple-boards-udev <https://api.riot-os.org/advanced-build-system-tricks.html#multiple-boards-udev>`__
+and use an easy to remember variable to identify BOARDs (which would
+allow also running the same python code on different setups), if
+following the above guide:
+
+.. code:: python
+
+   # first samr21-xpro
+   env1 = {'BOARD': 'samr21-xpro', 'BOARD_NUM': 0}
+   ctrl1 = RIOTCtrl(env=env1, application_directory='.')
+   # second samr21-xpro
+   env2 = {'BOARD': 'samr21-xpro', 'BOARD_NUM': 1}
+   ctrl2 = RIOTCtrl(env=env2, application_directory='.')
+
+Factories
+~~~~~~~~~
+
+The same tasks are done multiple times creating the object flashing it,
+starting the terminal and making sure its clean up. Once experiments
+grow and take over multiple ctrls this can become tedious, using a
+Factory together with a context manager can help with this.
+
+Going back to our example lets write a factory inheriting from
+``RIOTCtrlFactoryBase``:
+
+.. code:: python
+
+   from contextlib import ContextDecorator
+   from riotctrl.ctrl import RIOTCtrl, RIOTCtrlBoardFactory
+   from riotctrl_ctrl import native
+
+   class RIOTCtrlAppFactory(RIOTCtrlBoardFactory, ContextDecorator):
+
+       def __init__(self):
+           super().__init__(board_cls={
+               'native': native.NativeRIOTCtrl,
+           })
+           self.ctrl_list = list()
+
+       def __enter__(self):
+           return self
+
+       def __exit__(self, *exc):
+           for ctrl in self.ctrl_list:
+               ctrl.stop_term()
+
+       def get_ctrl(self, application_directory='.', env=None):
+           # retrieve a RIOTCtrl Object
+           ctrl = super().get_ctrl(
+               env=env,
+               application_directory=application_directory
+           )
+           # append ctrl to list
+           self.ctrl_list.append(ctrl)
```

Comment:
Same for this (+ maybe it should be a set, rather than a list?)

Reply:
Can we leave this for a code PR?